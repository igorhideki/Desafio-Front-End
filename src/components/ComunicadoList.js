import React from 'react';
import PropTypes from 'prop-types';
import ComunicadoItem from './ComunicadoItem';

const ComunicadoList = (props) => (
  <div>
    {props.comunicados.length > 0
      && <ul className="comunicados">
        {props.comunicados.map((comunicado) => (
          <ComunicadoItem key={comunicado.id} {...comunicado}/>
        ))}
      </ul>
    }
    {props.comunicados.length === 0 
      && <div className="comunicados__item">Nenhum comunicado encontrado.</div>
    }
  </div>
);

ComunicadoList.propTypes = {
  comunicados: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    seen: PropTypes.bool.isRequired,
    type: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    fullDescription: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
    file: PropTypes.shape({
      type: PropTypes.string,
      name: PropTypes.string,
      url: PropTypes.string,
    })
  }))
};

export default ComunicadoList;
