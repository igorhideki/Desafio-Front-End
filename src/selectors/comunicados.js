import moment from 'moment';

const getVisiblesComunicados = (comunicados, { type = -1 }) => {
  const formatDate = 'YYYY-MM-DD HH:mm';
  return comunicados.filter((comunicado) => {
    if (type === -1) {
      return comunicado;
    }
    return comunicado.type === type;
  }).sort((a, b) => {
    return moment(a.date, formatDate).format() < moment(b.date, formatDate).format() ? 1 : -1;
  });
};

export default getVisiblesComunicados;

