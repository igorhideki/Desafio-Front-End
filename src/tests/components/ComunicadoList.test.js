import React from 'react';
import { shallow } from 'enzyme';
import ComunicadoList from '../../components/ComunicadoList';
import comunicados from '../fixtures/comunicados';

test('should render ComunicadosList correctly with provided data', () => {
  const wrapper = shallow(<ComunicadoList comunicados={comunicados}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render ComunicadosList with empty data', () => {
  const wrapper = shallow(<ComunicadoList  comunicados={[]} />);
  expect(wrapper).toMatchSnapshot();
});
