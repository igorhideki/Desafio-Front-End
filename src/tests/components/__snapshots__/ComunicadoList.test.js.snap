// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ComunicadosList correctly with provided data 1`] = `
<div>
  <ul
    className="comunicados"
  >
    <withRouter()
      date="2018-05-03 8:30"
      description="Planos controle agora também disponível no cartão de crédito."
      fullDescription="Planos controle agora também disponível no cartão de crédito."
      id={1}
      key="1"
      seen={true}
      title="Metas Março"
      type={0}
    />
    <withRouter()
      date="2018-05-02 10:30"
      description="Planos controle 5GB de internet 4G."
      fullDescription="Ligações ilimitadas locais para todas as operadoras. Ligações ilimitadas pra fixos nacional e qualquer Vivo do Brasil com o código 15. SMS ilimitado para qualquer operadora do Brasil."
      id={2}
      key="2"
      seen={false}
      title="Bônus Fevereiro"
      type={1}
    />
    <withRouter()
      date="2018-05-02 18:30"
      description="Novos serviços, clique para visualizar o material de divulgação."
      files={
        Array [
          Object {
            "name": "Material de divulgação",
            "type": "image/*",
            "url": "logob4.png",
          },
        ]
      }
      fullDescription="Novos serviços, clique para visualizar o material de divulgação."
      id={3}
      key="3"
      seen={false}
      title="Novos Serviços"
      type={1}
    />
  </ul>
</div>
`;

exports[`should render ComunicadosList with empty data 1`] = `
<div>
  <div
    className="comunicados__item"
  >
    Nenhum comunicado encontrado.
  </div>
</div>
`;
